WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/OrderCustomerSection.swift

```diff
@@ -7,22 +7,26 @@ struct OrderCustomerSection: View {
     /// Parent view model to access all data
     @ObservedObject var viewModel: NewOrderViewModel
 
-    /// View model to drive the view content
-    private var customerDataViewModel: NewOrderViewModel.CustomerDataViewModel {
-        viewModel.customerDataViewModel
-    }
+    /// View model for the address form.
+    ///
+    @ObservedObject var addressFormViewModel: CreateOrderAddressFormViewModel
 
     @State private var showAddressForm: Bool = false
 
     var body: some View {
         OrderCustomerSectionContent(viewModel: viewModel.customerDataViewModel, showAddressForm: $showAddressForm)
-        .sheet(isPresented: $showAddressForm) {
-            NavigationView {
-                EditOrderAddressForm(dismiss: {
-                    showAddressForm.toggle()
-                }, viewModel: viewModel.createOrderAddressFormViewModel())
+            .sheet(isPresented: $showAddressForm) {
+                NavigationView {
+                    EditOrderAddressForm(dismiss: {
+                        showAddressForm.toggle()
+                    }, viewModel: addressFormViewModel)
+                }
+                .discardChangesPrompt(canDismiss: !addressFormViewModel.hasPendingChanges,
+                                      didDismiss: addressFormViewModel.userDidCancelFlow)
+                .onDisappear {
+                    viewModel.addressFormViewModel = viewModel.createOrderAddressFormViewModel()
```

Comment:
One minor nit, I would have created a method called `viewModel.resetAddressForm()` to hide this complexity. Then we can add comments on why is this needed.